vm/src/pyobject.rs

```diff
@@ -642,9 +643,8 @@ pub struct PyObject<T>
 where
     T: ?Sized + PyObjectPayload,
 {
-    pub typ: Arc<PyObject<PyClass>>,
-    // TODO: make this RwLock once PyObjectRef is Send + Sync
-    pub(crate) dict: Option<Mutex<PyDictRef>>, // __dict__ member
+    pub(crate) typ: ArcSwap<PyObject<PyClass>>, // __class__ member
+    pub(crate) dict: Option<ArcSwap<PyObject<PyDict>>>, // __dict__ member
```

Comment:
Very useful type! I was thinking about using a sort of `AtomicArcCell` for these fields, but I wasn't sure it would be able to be fully atomic; looks like it is!

Reply:
It suits out use case pretty well as it's optimised for reading, which is the operation that we really care about here.